enhancements/dns/dns-operator-operand-logging-level.md

```diff
@@ -0,0 +1,228 @@
+We will using [logrus](https://github.com/sirupsen/logrus#level-logging) to enable the log levels for the operator level logging.
+Logrus has seven logging levels: Trace, Debug, Info, Warning, Error, Fatal and Panic.
+log.Trace("Something very low level.")
+log.Debug("Useful debugging information.")
+log.Info("Something noteworthy happened!")
+log.Warn("You should probably take a look at this.")
+log.Error("Something failed but I'm not quitting.") // Calls os.Exit(1) after logging log.Fatal("Bye.") // Calls panic() after logging log.Panic("I'm bailing.")
+After the logging level on a Logger is set, log entries with that severity or anything above it will be logged.
+Eg- Default log.SetLevel(log.InfoLevel) will log anything that is info or above (warn, error, fatal, panic).
+So, we will be setting operatorLogLevel in a separate controller to watch dnses and set log level.
+operatorLogLevel: "Normal" will set logrus.SetLogLevel("Info")
+operatorLogLevel: "Debug" will set logrus.SetLogLevel("Debug")
+operatorLogLevel: "Trace" will set  logrus.SetLogLevel("Trace")
+
+
+### CoreDNS Log Level API
+
+Valid values for logLevel are: "Normal", "Debug", "Trace" as per [Operator Api](https://github.com/openshift/api/blob/master/operator/v1/types.go#L62)
+
+We will enable logging of CoreDNS's [classes of responses](https://github.com/coredns/coredns/tree/master/plugin/log#syntax) that correspond to the log level specified in the API.
+So,
+
+logLevel "Normal"  will enable errors class as we enable errors plugin by default log . { class error }
+
+logLevel "Debug" will enable log . { class denial error }
+
+logLevel "Trace" will enable  log . { class all }
+
+
+
+We will creating an API under DNSSpec with **field name** `operatorlogLevel` of type [LogLevel](https://github.com/openshift/api/blob/master/operator/v1/types.go#L71)
+
+```go
+type DNSSpec struct {
+	// <snip>
+
+	// operatorLogLevel controls the logging level of the DNS Operator.
+	// See LogLevel for more information about each available logging level.
+	//
+	// +optional
+	OperatorLogLevel LogLevel `json:"operatorLogLevel"`
+}
+```
+
+This new field would allow a cluster administrator to specify the desired logging level specifically for the DNS Operator.
+
+Additionally, a new `LogLevel` of type [LogLevel](https://github.com/openshift/api/blob/master/operator/v1/types.go#L62) will be added for CoredDNS logging :
+
+```go
+// We will be enabling the CLASSES(https://github.com/coredns/coredns/tree/master/plugin/log#syntax) of coredns w.r.t to the LogLevel we have defined in openshift api.
+
+// So,
+
+// logLevel "Normal"  will enable errors class as we enable errors plugin by default log . { class error }
+
+// logLevel "Debug" will enable log . { class denial error }
+
+// logLevel "Trace" will enable  log . { class all }
+type DNSSpec struct {
+	
+	// logLevel describes the logging verbosity of the DNSController for CoreDNS.
+	//
+	// +optional
+	LogLevel LogLevel `json:"logLevel"`
```

Comment:
add defaulting to the standard loglevel.

Reply:
sorry @sttts forgot to update here as had added it the api https://github.com/openshift/api/pull/1031
@sttts just a request if would be able to review https://github.com/openshift/api/pull/1031 then that will be great.

Reply:
@sttts PTAL just added the changes you requested. Thanks in advance!